qiskit/algorithms/minimum_eigensolvers/adapt_vqe.py

```diff
@@ -294,10 +294,10 @@ class AdaptVQEResult(VQEResult):
 
     def __init__(self) -> None:
         super().__init__()
-        self._num_iterations: int = None
-        self._final_max_gradient: float = None
+        self._num_iterations: int | None = None
+        self._final_max_gradient: float | None = None  # TODO: should be complex?
```

Comment:
From other places, it appears that `ma_gradient` is a complex variable

Reply:
This variable is the energy gradient, so it should be real. Maybe adding a `.real` in the `_compute_gradient` function would resolve the ambiguity?

Reply:
So here
https://github.com/Qiskit/qiskit-terra/blob/aa3320ced0a669fc9d9ae46632071d7134a8f119/qiskit/algorithms/observables_evaluator.py#L29-L35
the return type should be float? In particular, `EstimatorResult` value field is `np.ndarray` of floats, but for some reason it is considered complex in this point.

Reply:
There the return value could indeed be complex, e.g. if a user passes an observable with complex coefficient, such as
```python
SparsePauliOp(["Z"], coeffs=[2j])
```
But in the `AdaptVQE` we know the `hamiltonian` must have real energies, otherwise we couldn't minimize it correctly 🙂